### ISSUE TITLE
Removing image crate from wasm builds

### ISSUE THREAD

**Issue by valaphee** [bevyengine/bevy#13247]
## What problem does this solve or what need does it fill?

There is no need for a full-on image parsing library on wasm builds, as the user agent integrates plenty of image parser. It's also possible to directly initialize images from such sources. Either through `HTMLImageElement` or `ImageData`.

This would improve the loading times, by removing the image parser running in wasm, and the delivery time of the wasm file itself, as I assume that the image crate itself is quite heavy.

## What solution would you like?

Switching over to using `HTMLImageElement`, but this would require to somehow prevent the AssetServer from fetching, and the AssetLoader getting the full url to create an `HTMLImageElement` which is handed over to `wgpu`.

## Additional context

Requires https://github.com/gfx-rs/wgpu/pull/5668

**Comment by mnmaita** [comment]
Looks like https://github.com/gfx-rs/wgpu/pull/5668 got merged

**Comment by valaphee** [comment]
Yup in itself this is pretty simple to implement, but images are stored as HTMLImageElement, where the data cannot easily be retrieved (e.g. is available)

And the wasm asset loader should fetch files lazily e.g. VecReader gets replaced with "WebReader" and the first read triggers a fetch, or uses the streaming functionalities. (would also make asset loading faster in general, as while the asset is being downloaded it can already be parsed)
